Amazon.QLDB.Driver.IntegrationTests/AsyncStatementExecutionTests.cs

```diff
@@ -0,0 +1,1053 @@
+    using Microsoft.VisualStudio.TestTools.UnitTesting;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+    
+    [TestClass]
+    public class AsyncStatementExecutionTests
+    {
+        private static readonly IValueFactory ValueFactory = new ValueFactory();
+        private static AmazonQLDBSessionConfig amazonQldbSessionConfig;
+        private static IntegrationTestBase integrationTestBase;
+        private static AsyncQldbDriver qldbDriver;
+
+        [ClassInitialize]
+        public async static Task SetUp(TestContext context)
+        {
+            // Get AWS configuration properties from .runsettings file.
+            string region = context.Properties["region"].ToString();
+            
+            amazonQldbSessionConfig = IntegrationTestBase.CreateAmazonQLDBSessionConfig(region);
+            integrationTestBase = new IntegrationTestBase(Constants.LedgerName, region);
+
+            integrationTestBase.RunForceDeleteLedger();
+
+            integrationTestBase.RunCreateLedger();
+            qldbDriver = integrationTestBase.CreateAsyncDriver(amazonQldbSessionConfig);
+
+            // Create table.
+            var query = $"CREATE TABLE {Constants.TableName}";
+
+            var count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+
+
+            Assert.AreEqual(1, count);
+
+            var result = await qldbDriver.ListTableNamesAsync();
+            foreach (var row in result)
+            {
+                Assert.AreEqual(Constants.TableName, row);
+            }
+        }
+
+        [ClassCleanup]
+        public static void ClassCleanup()
+        {
+            integrationTestBase.RunDeleteLedger();
+            qldbDriver.Dispose();
+        }
+
+        [TestCleanup]
+        public async Task TestCleanup()
+        {
+            // Delete all documents in table.
+            await qldbDriver.Execute(async txn => await txn.Execute($"DELETE FROM {Constants.TableName}"));
+        }
+
+        [TestMethod]
+        public async Task Execute_DropExistingTable_TableDropped()
+        {
+            // Given.
+            var create_table_query = $"CREATE TABLE {Constants.CreateTableName}";
+            var create_table_count = await qldbDriver.Execute(async txn =>
+            {
+                var result = await txn.Execute(create_table_query);
+
+                var count = 0;
+                await foreach (var row in result)
+                {
+                    count++;
+                }
+                return count;
+            });
+            Assert.AreEqual(1, create_table_count);
+
+            // Execute ListTableNames() to ensure table is created.
```

Comment:
ListTableNamesAsync()

Reply:
https://github.com/justing-bq/amazon-qldb-driver-dotnet/pull/9 actually renames the method to `ListTableNames()` so here we will leave this comment alone and update the actual function call.